libs/data-types/src/lib/utilities/bootstrap-dynamic-types/resolve-member-schemas-for-union.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { ThingDataOne, ThingDataTwo, Widget } from '../../../test/widget';
+import {
+    NestedDataType,
+    NonNegativeFiniteNumber,
+    Union2,
+    Union2Member,
+    UUID,
+} from '../../decorators';
+import {
+    resolveMemberSchemasForUnion,
+    UnionMemberSchemaDefinition,
+} from './resolve-member-schemas-for-union';
+
+const assertMembersResolved = (
+    schemaDefinitions: UnionMemberSchemaDefinition[],
+    expectedNumberOfMembers: number,
+    expectedDiscriminantValues: string[]
+): void => {
+    expect(schemaDefinitions.length).toBe(expectedNumberOfMembers);
+
+    const missingMembers = expectedDiscriminantValues.filter(
+        (discriminantValue) =>
+            !schemaDefinitions.some(
+                ({ discriminantValue: discriminantValueForThisSchema }) =>
+                    discriminantValue === discriminantValueForThisSchema
+            )
+    );
+
+    expect(missingMembers).toEqual([]);
+};
+
+describe(`resolveMemberSchemasForUnion`, () => {
+    describe(`when there are multiple union members with no nesting`, () => {
+        it(`should resolve the members`, () => {
+            const result = resolveMemberSchemasForUnion(
+                [Widget, ThingDataOne, ThingDataTwo],
+                'THING_UNION'
+            );
+
+            assertMembersResolved(result, 2, ['one', 'two']);
+        });
+    });
+
+    describe(`when the one of the union members leverages a nested type`, () => {
```

Comment:
when the one of the union members -- when one of the union members

Reply:
Good catch!

Reply:
fixed